api/v1/helpers/verbs/utils.js

```diff
@@ -22,6 +22,75 @@ const publisher = require('../../../../realtime/redisPublisher');
 const realtimeEvents = require('../../../../realtime/constants').events;
 
 /**
+ * @param {Object} o Sequelize instance
+ * @param {Object} puttableFields from API
+ * @param {Object} toPut from request.body
+ * @returns {Promise} the updated instance
+ */
+function updateInstance(o, puttableFields, toPut) {
+  const keys = Object.keys(puttableFields);
+  for (let i = 0; i < keys.length; i++) {
+    const key = keys[i];
+    if (toPut[key] === undefined) {
+      let nullish = null;
+      if (puttableFields[key].type === 'boolean') {
+        nullish = false;
+      } else if (puttableFields[key].enum) {
+        nullish = puttableFields[key].default;
+      }
+
+      o.set(key, nullish);
+
+      // take nullified fields out of changed fields
+      o.changed(key, false);
+    } else {
+
+      /*
+       * value may have changed. set changed to true to
+       * trigger checks in the model
+       */
+      o.changed(key, true);
+      o.set(key, toPut[key]);
+    }
+  }
+
+  return o.save();
+}
+
+/**
+ * Sends the udpated record back in the json response
+ * with status code 200.
+ *
+ * @param {Object} resultObj - For logging
+ * @param {Object} req - The request object
+ * @param {Object} retVal - The updated instance
+ * @param {Object} props - The helpers/nouns module for the given DB model
+ * @param {Object} res - The response object
+ * @returns {Object} JSON succcessful response
+ */
+function handleUpdatePromise(resultObj, req, retVal, props, res) {
+  resultObj.dbTime = new Date() - resultObj.reqStartTime;
+  logAPI(req, resultObj, retVal);
+
+  // publish the update event to the redis channel
+  if (props.publishEvents) {
+    publisher.publishSample(
+      retVal, props.associatedModels.subject, realtimeEvents.sample.upd);
+  }
+
+  // update the cache
+  if (props.cacheEnabled) {
+    const getCacheKey = req.swagger.params.key.value;
+    const findCacheKey = '{"where":{}}';
+    redisCache.del(getCacheKey);
+    redisCache.del(findCacheKey);
```

Comment:
redisCache is not defined. Wondering if there are tests for these ?

Reply:
added redisCache to utils file. The handleUpdatePromise function is called from tests expecting status code 200 in tests/api/v1/generators/putWithCollector.js.